dev/RadioButtons/RadioButtons.cpp

```diff
@@ -163,63 +455,206 @@ void RadioButtons::OnPropertyChanged(const winrt::DependencyPropertyChangedEvent
+    {
+        m_selectionModel.ClearSelection();
+    }
 }
 
 void RadioButtons::UpdateSelectedIndex()
 {
-    if (auto listView = m_listView.get())
+    m_selectionModel.Select(SelectedIndex());
+}
+
+winrt::UIElement RadioButtons::ContainerFromItem(winrt::IInspectable const& item)
+{
+    if (auto const repeater = m_repeater.get())
     {
-        listView.SelectedIndex(SelectedIndex());
+        if (auto const itemTemplate = repeater.ItemTemplate())
+        {
+            if (auto const itemFactory = itemTemplate.try_as<winrt::RadioButtonsElementFactory>())
+            {
+                return itemFactory.ContainerFromItem(item);
+            }
+        }
     }
+    return nullptr;
 }
 
-winrt::DependencyObject RadioButtons::ContainerFromItem(winrt::IInspectable const& item)
+winrt::UIElement RadioButtons::ContainerFromIndex(int index)
 {
-    if (auto listView = m_listView.get())
+    if (auto const repeater = m_repeater.get())
     {
-        return listView.ContainerFromItem(item);
+        return repeater.TryGetElement(index);
     }
     return nullptr;
 }
 
-winrt::DependencyObject RadioButtons::ContainerFromIndex(int index)
+// Test Hooks helpers, only function when m_testHooksEnabled == true
+void RadioButtons::SetTestHooksEnabled(bool enabled)
+{
+    if (m_testHooksEnabled != enabled)
+    {
+        m_testHooksEnabled = enabled;
+        if (enabled)
+        {
+            AttachToLayoutChanged();
+        }
+        else
+        {
+            DetatchFromLayoutChanged();
+        }
+    }
+}
+
+RadioButtons::~RadioButtons()
+{
+    if (m_layoutChangedToken)
+    {
+        if (auto const layout = GetLayout())
+        {
+            layout->LayoutChanged(m_layoutChangedToken);
+            m_layoutChangedToken = { 0 };
+        }
+    }
+}
+
+void RadioButtons::OnLayoutChanged(const winrt::ColumnMajorUniformToLargestGridLayout&, const winrt::IInspectable&)
+{
+    RadioButtonsTestHooks::NotifyLayoutChanged(*this);
+}
+
+int RadioButtons::GetRows()
+{
+    if (auto const layout = GetLayout())
+    {
+        return layout->GetRows();
+    }
+    return -1;
+}
+
+int RadioButtons::GetColumns()
+{
+    if (auto const layout = GetLayout())
+    {
+        return layout->GetColumns();
+    }
+    return -1;
+}
+
+int RadioButtons::GetLargerColumns()
+{
+    if (auto const layout = GetLayout())
+    {
+        return layout->GetLargerColumns();
+    }
+    return -1;
+}
+
+
+void RadioButtons::AttachToLayoutChanged()
+{
+    if (auto const layout = GetLayout())
+    {
+        layout->SetTestHooksEnabled(true);
+        m_layoutChangedToken = layout->LayoutChanged({ this, &RadioButtons::OnLayoutChanged });
+    }
+}
+
+void RadioButtons::DetatchFromLayoutChanged()
+{
+    if (auto const layout = GetLayout())
+    {
+        layout->SetTestHooksEnabled(false);
+        layout->LayoutChanged(m_layoutChangedToken);
+        m_layoutChangedToken = { 0 };
+    }
+}
+
+ColumnMajorUniformToLargestGridLayout* RadioButtons::GetLayout()
```

Comment:
>ColumnMajorUniformToLargestGridLayout* [](start = 0, length = 38)

This should probably return a com_ptr<ColumnMajorUniformToLargestGridLayout> instead of raw pointer